src/Auth/Source/OAuth2.php

```diff
@@ -71,26 +74,26 @@ public function __construct(array $info, array $config)
 
         // Call the parent constructor first, as required by the interface
         parent::__construct($info, $config);
-        if (array_key_exists('template', $config)) {
-            $template = $config['template'];
-            if (is_string($template)) {
-                $templateArray = constant(ConfigTemplate::class . '::' . $template);
+        if (\array_key_exists('template', $config)) {
+            if (\is_string($config['template'])) {
+                /** @var array $templateArray */
+                $templateArray = \constant(ConfigTemplate::class . '::' . $config['template']);
                 // Remove the class name
                 unset($templateArray[0]);
                 $config = array_merge($templateArray, $config);
             }
         }
-        if (!array_key_exists('redirectUri', $config)) {
-            $config['redirectUri'] = Module::getModuleURL('authoauth2/linkback.php');
```

Comment:
@ioigoume  The redirect URIs is something that users of this module of already registered with Facebook, Google, etc. I suppose it is like an ACS url.

In Cirrus's case we can't change the redirect URI.  We don't have access to the Facebook/Google/LinkedIn apps to make the change.

So in our case we need it to still end with `.php` and to have route back correctly.

Question for you:   should we only support urls `.php` endings?  or support both? If both which should be the default?   I don't recall how many different urls get registered, but there are also OIDC logout urls.

Reply:
@pradtke ,
i added a new configuration: `'useLegacyRoutes' => true,` . If present and true, the module will use the legacy endpoints, meaning all endpoints will end with `.php`. If not present or false it will use the new ones.

Please give it a try and let me know if it works as expected for you